File: arrays/closestToZero.cpp
#include <iostream>
#include <vector>
#include <cmath>

int closestToZero(std::vector<int>& arr){
    bool firstRun = true;
    int closest = 0;
    int current = 0;

    for(int i = 0;i < arr.size();i++){
        if(firstRun){
            closest = arr[i];
            firstRun = false;
        }

        if(abs(arr[i]) < abs(closest) || (abs(arr[i]) == abs(closest) && arr[i] > closest)){
            closest = arr[i];
        }

    }

    return closest;
}

File: arrays/closestToZero.py
def closestToZero(arr):
    closestToZero = None
    
    for n in arr:
        if(closestToZero is None):
            closestToZero = n
            continue

        if(abs(n) == abs(closestToZero) and n != closestToZero):
            closestToZero = abs(n)
            continue

        if(abs(n) < abs(closestToZero)):
            closestToZero = n
    
    return closestToZero


arr = [-4,-2,14,-1,4,8]
print(closestToZero(arr))